### ISSUE TITLE
Export Issue Vertex Colors / Alpha

### ISSUE THREAD

**Issue by SilentiCo** [niftools/blender_nif_plugin#358]
@niftools/blender-nif-plugin-reviewer -

# Issue Overview
This is maybe a personal issue do to lack of knowledge! However I wasn't able to find any proper help behind the offical readme and some less usefull forum posts.

**Error while export a mesh with vertex colors / vertex alpha:**
**"Attribute Error: 'bpy_prop_array' object has no attribute 'r' "**  
Export failed as soon as a vertex color layer is added and painted. In the Material Tab vertex color paint is enabled. Under object tab NIF TOOL SHADERS vertex color is enabled.

### Blender Nif Plugin Version Info
blender_nif_plugin-2.6.0a0.dev4-02.04.2016
(With newer versions other issues accure)

### Blender Version Info
v2.79.7

### Platform information
Win 8.1

## Steps to Reproduce
1. Create mesh
2. add material
3. add texture, set uv, add normal map
4. UV unwrap
5. Add shader, triangulate modifier
6. seperate by material
7. TEST Export all works fine
8. Enable vertex color in shader options, enable vertex color paint under material tab
9. Add vertex color to mesh
10. Export fails

## Expected Result
Export mesh with vertex colors or vertex alpha similar to the solitude meshes (skyrim).
Technique often used by Bethesda.

## Actual Result
Export failed. 

## Possible Fix
1. Missing steps for proper export.
2. feature not working in current version of Nif Tools or Blender version

## Logs and Files
[Provide logs file generated during the error as well as the blend and nif files are related to the issue]

### Info Bar Output
Only a small "red" section

bpy.data.window_managers["WinMan"].(null) = 100
Blender NIF Scripts 2.6.0 (running on Blender 2.79 (sub 7), PyFFI 2.2.2)
Traceback (most recent call last):
  File "C:\Users\xxxxxxx\AppData\Roaming\Blender Foundation\Blender\2.79\scripts\addons\io_scene_nif\operators\nif_export_op.py", line 171, in execute
    return nif_export.NifExport(self, context).execute()
  File "C:\Users\xxxxxxx\AppData\Roaming\Blender Foundation\Blender\2.79\scripts\addons\io_scene_nif\nif_export.py", line 261, in execute
    self.objecthelper.export_node(root_object, 'localspace', root_block, root_object.name)
  File "C:\Users\xxxxxxx\AppData\Roaming\Blender Foundation\Blender\2.79\scripts\addons\io_scene_nif\objectsys\object_export.py", line 214, in export_node
    self.mesh_helper.export_tri_shapes(b_obj, space, parent_block, node_name)
  File "C:\Users\xxxxxxx\AppData\Roaming\Blender Foundation\Blender\2.79\scripts\addons\io_scene_nif\objectsys\object_export.py", line 977, in export_tri_shapes
    fcol = [b_color.r, b_color.g, b_color.b, 1.0]
AttributeError: 'bpy_prop_array' object has no attribute 'r'

location: <unknown location>:-1

### Console Output
Attribute Error: 'bpy_prop_array' object has no attribute 'r'

**Comment by neomonkeus** [comment]
You are using an outdated version of the plugin.
This was fixed by #342 Update documentation for 2.8 + mat sys improvements

**Comment by SilentiCo** [comment]
Hi, 
first of all thank you for your (very) fast reply!

I tried using the new 2.8 Nif tools version...I am however completly unfamilar with the new ways of doing.
Especialy what render sytem to use ( Eevee, Cycles or Workbench), how to setup the texture itself via notes? Principled BSDF etc.  I am aware that updating and keeping these scripts a life is already alot of work but is there a up to date tutorial for these tools? 

I tried setting up texture, material (Principled BSDF), game version etc. as usual but get this error while exporting:
**Attribute Error: 'ObjectProperty' object has no attribute 'bss_helper'**  no clue what that could possibly mean XD.

Greetings!
